samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -0,0 +1,68 @@
+"""
+Custom Click options for hook package id
+"""
+
+import logging
+import os
+import click
+
+from samcli.lib.hook.exceptions import InvalidHookWrapperException
+from samcli.lib.hook.hook_wrapper import IacHookWrapper, get_available_hook_packages_ids
+
+LOG = logging.getLogger(__name__)
+
+
+class HookPackageIdOption(click.Option):
+    """
+    A custom option class that allows do custom validation for the SAM CLI commands options in case if hook package
+    id is passed. It also calls the correct IaC prepare hook, and update the SAM CLI commands options based on the
+    prepare hook output.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._force_prepare = kwargs.pop("force_prepare", True)
+        self._invalid_coexist_options = kwargs.pop("invalid_coexist_options", [])
+        super().__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        opt_name = self.name.replace("_", "-")
+        if self.name in opts:
+            # validate the hook_package_id value exists
+            hook_package_id = opts[self.name]
+            iac_hook_wrapper = None
+            try:
+                iac_hook_wrapper = IacHookWrapper(hook_package_id)
+            except InvalidHookWrapperException as e:
+                raise click.BadParameter(
+                    f"{hook_package_id} is not a valid hook package id. This is the list of valid "
+                    f"packages ids {get_available_hook_packages_ids()}"
+                ) from e
+
+            # validate coexist options
+            for invalid_opt in self._invalid_coexist_options:
+                invalid_opt_name = invalid_opt.replace("-", "_")
+                if invalid_opt_name in opts:
+                    raise click.BadParameter(
+                        f"Parameters {opt_name}, and {','.join(self._invalid_coexist_options)} could not be used "
```

Comment:
I thought it is better to let the customer knows what are the options that should not be used with the hook package id instead of just letting them know that they are using a wrong option

Reply:
Ah yeah I think you're right. Nit: could we change it `could not` to `can not` 🙏